torch_dicom/inference/pipeline.py

```diff
@@ -0,0 +1,320 @@
+import logging
+from abc import ABC, abstractmethod
+from copy import copy
+from dataclasses import dataclass, field
+from functools import partial
+from pathlib import Path
+from typing import Any, Dict, Iterable, Iterator, List, Optional, Sequence, Tuple, TypeVar, Union
+
+import torch
+import torch.nn as nn
+from dicom_utils.dicom import Dicom
+from dicom_utils.volume import ReduceVolume, VolumeHandler
+from PIL import Image
+from torch import Tensor
+from torch.utils.data import ChainDataset, ConcatDataset, DataLoader, Dataset
+from torchvision.transforms.v2 import Compose
+from tqdm import tqdm
+
+from torch_dicom.datasets import (
+    DicomInput,
+    DicomPathDataset,
+    DicomPathInput,
+    ImageInput,
+    ImagePathDataset,
+    ImagePathInput,
+    TensorInput,
+    TensorPathDataset,
+    TensorPathInput,
+    collate_fn,
+    uncollate,
+)
+from torch_dicom.datasets.dicom import Transform
+from torch_dicom.preprocessing import MinMaxCrop, Resize
+
+
+B = TypeVar("B")
+
+
+@dataclass(repr=False)
+class InferencePipeline(ABC):
```

Comment:
The lightning functionality is separated from the base class to avoid forcing lightning as a dependency when using this pipeline (it's still an optional extra). I could also see a need for non-lightning inference pipelines in the future, mostly because lightning is a heavy dependency and will not always be chosen over alternatives (huggingface accelerate, etc.)